File: 3011_find_if_array_can_be_sorted.cpp
class Solution {
public:
    bool canSortArray(vector<int>& nums) {
        bool ongoing = true;
        while (ongoing) {
            ongoing = false;
            for (int i = 1; i < nums.size(); i++) {
                if (__builtin_popcount(nums[i]) == __builtin_popcount(nums[i-1]) && nums[i-1] > nums[i]) {
                    swap(nums[i], nums[i-1]);
                    ongoing = true;
                }
            }
        }
        return is_sorted(nums.begin(), nums.end());
    }
};

File: 3011_find_if_array_can_be_sorted.java
class Solution {
    public boolean canSortArray(int[] nums) {
        boolean sortFlag = true;
        while (sortFlag) {
            sortFlag = false;
            for (int k = 1; k < nums.length; k++) {
                if (Integer.bitCount(nums[k]) == Integer.bitCount(nums[k - 1]) && nums[k - 1] > nums[k]) {
                    int tmp = nums[k];
                    nums[k] = nums[k - 1];
                    nums[k - 1] = tmp;
                    sortFlag = true;
                }
            }
        }
        return sorted(nums);
    }

    private boolean sorted(int[] nums) {
        for (int k = 1; k < nums.length; k++) {
            if (nums[k - 1] > nums[k]) return false;
        }
        return true;
    }
}
